routing/router_test.go

```diff
@@ -647,6 +647,149 @@ func TestSendPaymentErrorRepeatedFeeInsufficient(t *testing.T) {
+	errChanUpdate := lnwire.ChannelUpdate{
+		ShortChannelID: lnwire.NewShortChanIDFromInt(privateChannelID),
+		Timestamp:      uint32(testTime.Add(time.Minute).Unix()),
+		BaseFee:        updatedFeeBaseMSat,
+		FeeRate:        updatedFeeProportionalMillionths,
+		TimeLockDelta:  expiryDelta,
+	}
+
+	err = signErrChanUpdate(ctx.privKeys["songoku"], &errChanUpdate)
+	if err != nil {
+		t.Fatalf("Failed to sign channel update error: %v ", err)
+	}
+
+	// The error will be returned by son goku, before forwarding to the
+	// private channel, to simulate an increase in fee on the private
+	// channel that wasn't propagated back to the network.
+	sourceNode := ctx.aliases["songoku"]
+
+	// We'll now modify the SendToSwitch method to return an error on
+	// the first attempt, and check that the correct route and fees
+	// where used the second attempt.
+	errorReturned := false
+	ctx.router.cfg.SendToSwitch = func(firstHop lnwire.ShortChannelID,
+		_ *lnwire.UpdateAddHTLC, _ *sphinx.Circuit) ([32]byte, error) {
+
+		if !errorReturned {
+			errorReturned = true
+			return [32]byte{}, &htlcswitch.ForwardingError{
+				ErrorSource: sourceNode,
+
+				// Within our error, we'll add a channel update
+				// which is meant to reflect the new fee
+				// schedule for the node/channel.
+				FailureMessage: &lnwire.FailFeeInsufficient{
+					Update: errChanUpdate,
+				},
+			}
+		}
+
+		return preImage, nil
+	}
+
+	// Avoid dropping the channel information upon check if the
+	// target channel is still open by querying the utxo-set for
+	// its existence.
+	ctx.router.cfg.AssumeChannelValid = true
+
+	// Send off the payment request to the router, route through son
+	// goku and then across the private channel to elst.
+	paymentPreImage, route, err := ctx.router.SendPayment(&payment)
+	if err != nil {
+		t.Fatalf("unable to send payment: %v", err)
+	}
+
+	// The route selected should have two hops. Make sure that
+	// the son goku -> sophon -> elst or the pham nuwen -> sophon
+	// -> elst are not selected instead.
+	if len(route.Hops) != 2 {
+		t.Fatalf("incorrect route length: expected %v got %v", 2,
+			len(route.Hops))
+	}
+
+	// The preimage should match up with the one created above.
+	if !bytes.Equal(paymentPreImage[:], preImage[:]) {
+		t.Fatalf("incorrect preimage used: expected %x got %x",
+			preImage[:], paymentPreImage[:])
+	}
+
+	// The route should have son goku as the first hop.
+	if !bytes.Equal(route.Hops[0].PubKeyBytes[:],
+		ctx.aliases["songoku"].SerializeCompressed()) {
+
+		t.Fatalf("route should go through son goku as first hop, "+
+			"instead passes through: %v",
+			getAliasFromPubKey(route.Hops[0].PubKeyBytes[:],
+				ctx.aliases))
+	}
+
+	// The route should pass via the private channel.
+	if !route.containsChannel(privateChannelID) {
+		t.Fatalf("route did not pass through private channel " +
+			"between pham nuwen and elst")
+	}
+
+	// The route should have the updated fee.
```

Comment:
could also check the value of `errorReturned` here.

Reply:
I didn't understand this one.

Reply:
You can check that `errorReturned == true`.

Reply:
Ah. Added